lib/AST/GenericEnvironment.cpp

```diff
@@ -365,6 +365,7 @@ GenericEnvironment::getOrCreateArchetypeFromInterfaceType(Type depType) {
   if (rootGP->isParameterPack()) {
     assert(getKind() == Kind::Primary);
     result = PackArchetypeType::get(ctx, this, requirements.anchor,
+                                    requirements.packShape,
```

Comment:
Should we map the packShape into the generic environment to make it distinct for distinct archetypes? (We also map the superclass type into context)